src/auth/authService.ts

```diff
@@ -27,27 +35,67 @@ export class AuthService {
         name,
         surname,
         email,
-        hashpassword
+        hashpassword,
+        code
       );
+      await this.mail.sendMail(user.rows[0].id, email, code);
       return user;
     }
   }
 
   async loginUser(email: string, password: string) {
     let hashpassword = createHash("sha256").update(password).digest("hex");
     let userLogin = await this.repository.findUserByEmail(email);
-    if (userLogin.password === hashpassword) {
-      return {
-        token: jsonwebtoken.sign(
-          {
-            id: userLogin.id,
-            access_type: userLogin.access_type,
-          },
-          "secret"
-        ),
-      };
+    const isVerified = await this.repository.isVerified(userLogin.id);
+    if (isVerified) {
+      if (userLogin.password === hashpassword) {
+        return {
+          token: jsonwebtoken.sign(
+            {
+              id: userLogin.id,
+              access_type: userLogin.access_type,
+            },
+            "secret"
+          ),
+        };
+      } else {
+        throw new WrongCredentials();
+      }
     } else {
-      throw new WrongCredentials();
+      throw new EmailNotVerifies();
+    }
+  }
+
+  async verifyEmail(code: number, user_id: number){
+    let checkCode = await this.repository.getCode(user_id);
+    if(checkCode === code){
+      return this.repository.changeEmailIsVerified(user_id);
     }
+    else{
+      throw new WrongCode();
+    }
+  }
+
+  async passwordReset(email: string){
+    const code = randomInt(100000, 999999);
+    const findUser = await this.repository.findUserByEmail(email);
+    if (findUser) {
+      await this.mail.passwordReset(email, code);
```

Comment:
if u have two or more async function. U can use `Promise.all`, so they will calls paralelno, but not poslidovno
```js
await Promise.all([
  this.mail.passwordReset(email, code),
  this.repository.addCode(code, findUser.id)
])
```